packages/app-project/src/screens/SubjectTalkPage/components/SubjectTalkData/components/Discussions/components/StartDiscussionModal/StartDiscussionModalContainer.jsx

```diff
@@ -0,0 +1,161 @@
+import { arrayOf, bool, func, shape, string } from 'prop-types'
+import { useState } from 'react'
+
+import {
+  addQueryParams,
+  createComment,
+  createDiscussion,
+  fetchDiscussions
+} from '@helpers'
+
+import { useBoards } from '@hooks'
+
+import StartDiscussionModal from './StartDiscussionModal'
+
+function StartDiscussionModalContainer({
+  active = false,
+  discussions = [],
+  onClose,
+  projectId = '',
+  showCommentMessage = false,
+  subjectId = '',
+  userId = ''
+}) {
+  const [error, setError] = useState(null)
+  const [loading, setLoading] = useState(false)
+
+  const boardsQuery = {
+    page_size: 50,
+    section: `project-${projectId}`,
+  }
+
+  const {
+    data: boards,
+    isLoading: boardsLoading,
+    error: boardsError
+  } = useBoards(active ? boardsQuery : null)
+
+  function handleRouting(url) {
+    window.location.href = url
+  }
+
+  async function handleCreateComment(commentData) {
+    setLoading(true)
+    setError(null)
+    try {
+      const newComment = await createComment(commentData)
+      setLoading(false)
+      const url = `/projects/${newComment.project_slug}/talk/${newComment.board_id}/${newComment.discussion_id}?comment=${newComment.id}`
```

Comment:
`addQueryParams` doesn't work if the url argument includes a query param (i.e. `?comment=1234`), it duplicates the `?=` instead of including one with `&=`.